js/connection.js

```diff
@@ -51,6 +51,22 @@ weechat.factory('connection',
                     })
                 );
 
+                // After this init, weechat either does nothing and the connection stays open
+                // or weechat closes the connection of the password is wrong
+                // Give weechat the oportunity to close the connection before we send anything else
+                return new Promise(function(resolve, reject) {
+                    setTimeout(function() {
+                        //If it's still open after 100ms we'll assume we are connected
+                        if (ngWebsockets.readyState() === WebSocket.OPEN) {
+                            resolve();
+                        } else {
+                            reject();
+                        }
+                    },100);
```

Comment:
I think it should be documented that this is a bit of a magic number, but that it's there only to improve error communication